azurerm/resource_arm_network_interface.go

```diff
@@ -360,12 +371,16 @@ func resourceArmNetworkInterfaceRead(d *schema.ResourceData, meta interface{}) e
 		if configs[0].InterfaceIPConfigurationPropertiesFormat != nil {
 			privateIPAddress := configs[0].InterfaceIPConfigurationPropertiesFormat.PrivateIPAddress
 			d.Set("private_ip_address", *privateIPAddress)
+			privateIPAddressVersion := configs[0].InterfaceIPConfigurationPropertiesFormat.PrivateIPAddressVersion
+			d.Set("private_ip_address_version", string(privateIPAddressVersion))
 		}
 
 		addresses := make([]interface{}, 0)
 		for _, config := range configs {
 			if config.InterfaceIPConfigurationPropertiesFormat != nil {
```

Comment:
This might be a bit more readable if we:

```go
if ifProps := config.InterfaceIPConfigurationPropertiesFormat; ipProps != nil {
    if ifProps.PrivateIPAddressVersion == network.IPv4 {
```